src/code/z_actor.c

```diff
@@ -457,439 +1254,3573 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+    if ((actor->flags & ACTOR_FLAG_10000000) && (globalCtx->roomCtx.currRoom.enablePosLights || (MREG(93) != 0))) {
+        light->enablePosLights = true;
+    }
+
+    Lights_BindAll(light, globalCtx->lightCtx.listHead,
+                   (actor->flags & (ACTOR_FLAG_10000000 | ACTOR_FLAG_400000)) ? NULL : &actor->world.pos, globalCtx);
+    Lights_Draw(light, globalCtx->state.gfxCtx);
+
+    if (actor->flags & ACTOR_FLAG_1000) {
+        Matrix_SetStateRotationAndTranslation(
+            actor->world.pos.x + globalCtx->mainCamera.skyboxOffset.x,
+            actor->world.pos.y + ((actor->shape.yOffset * actor->scale.y) + globalCtx->mainCamera.skyboxOffset.y),
+            actor->world.pos.z + globalCtx->mainCamera.skyboxOffset.z, &actor->shape.rot);
+    } else {
+        Matrix_SetStateRotationAndTranslation(actor->world.pos.x,
+                                              actor->world.pos.y + (actor->shape.yOffset * actor->scale.y),
+                                              actor->world.pos.z, &actor->shape.rot);
+    }
+
+    Matrix_Scale(actor->scale.x, actor->scale.y, actor->scale.z, MTXMODE_APPLY);
+    Actor_SetObjectDependency(globalCtx, actor);
+
+    gSPSegment(POLY_OPA_DISP++, 0x06, globalCtx->objectCtx.status[actor->objBankIndex].segment);
+    gSPSegment(POLY_XLU_DISP++, 0x06, globalCtx->objectCtx.status[actor->objBankIndex].segment);
+
+    if (actor->colorFilterTimer != 0) {
+        s32 temp_v0_2 = actor->colorFilterParams & 0xC000;
+        Color_RGBA8 actorDefaultHitColor = { 0, 0, 0, 255 };
+
+        if (temp_v0_2 == 0x8000) {
+            actorDefaultHitColor.r = actorDefaultHitColor.g = actorDefaultHitColor.b =
+                ((actor->colorFilterParams & 0x1F00) >> 5) | 7;
+        } else if (temp_v0_2 == 0x4000) {
+            actorDefaultHitColor.r = ((actor->colorFilterParams & 0x1F00) >> 5) | 7;
+        } else if (temp_v0_2 == 0xC000) {
+            actorDefaultHitColor.b = actorDefaultHitColor.g = actorDefaultHitColor.r = 0;
+        } else {
+            actorDefaultHitColor.b = ((actor->colorFilterParams & 0x1F00) >> 5) | 7;
+        }
+
+        if (actor->colorFilterParams & 0x2000) {
+            func_800AE778(globalCtx, &actorDefaultHitColor, actor->colorFilterTimer, actor->colorFilterParams & 0xFF);
+        } else {
+            func_800AE434(globalCtx, &actorDefaultHitColor, actor->colorFilterTimer, actor->colorFilterParams & 0xFF);
+        }
+    }
+
+    actor->draw(actor, globalCtx);
+
+    if (actor->colorFilterTimer != 0) {
+        if (actor->colorFilterParams & 0x2000) {
+            func_800AE8EC(globalCtx);
+        } else {
+            func_800AE5A0(globalCtx);
+        }
+    }
+
+    if (actor->shape.shadowDraw != NULL) {
+        actor->shape.shadowDraw(actor, light, globalCtx);
+    }
+    actor->isDrawn = true;
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_800B9D1C(Actor* actor) {
+    s32 sfx = actor->sfx;
```

Comment:
```suggestion
    s32 sfxId = actor->sfxId;
```
You should be using `sfxId`, not `sfx` for temps